From d3800ba40ea75b92cf29b0cccb7bb08baa26dd36 Mon Sep 17 00:00:00 2001
From: Michael Scott <mike@foundries.io>
Date: Sun, 27 Mar 2022 16:49:44 -0700
Subject: [PATCH] imx8m: handle ENV_IS_IN_FAT in env_get_location

This fixes a u-boot hang when ENV_IS_NOWHERE is disabled and
ENV_IS_IN_FAT is enabled (but nothing else).

This can be reverted once merged to u-boot-fio.

Signed-off-by: Michael Scott <mike@foundries.io>
---
 arch/arm/mach-imx/imx8m/soc.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/imx8m/soc.c b/arch/arm/mach-imx/imx8m/soc.c
index f11921a5bb..4e9d4f5463 100644
--- a/arch/arm/mach-imx/imx8m/soc.c
+++ b/arch/arm/mach-imx/imx8m/soc.c
@@ -1399,14 +1399,18 @@ enum env_location env_get_location(enum env_operation op, int prio)
 		env_loc = ENVL_NAND;
 		break;
 #endif
-#ifdef CONFIG_ENV_IS_IN_MMC
+#if defined(CONFIG_ENV_IS_IN_MMC) || defined(CONFIG_ENV_IS_IN_FAT)
 	case SD1_BOOT:
 	case SD2_BOOT:
 	case SD3_BOOT:
 	case MMC1_BOOT:
 	case MMC2_BOOT:
 	case MMC3_BOOT:
+#ifdef CONFIG_ENV_IS_IN_MMC
 		env_loc =  ENVL_MMC;
+#else
+		env_loc =  ENVL_FAT;
+#endif
 		break;
 #endif
 	default:
-- 
2.35.1

